### PR TITLE
Added message and disabled the dropdown in email view when no participants are present

### PR DESCRIPTION
**Pull request by nikhilgupta1211**

### CODE REVIEW

File: app/views/event_emails/new.html.haml [hunk 1]
```diff
@@ -27,7 +27,17 @@
...
   function user_email(users){
-    var recip = document.getElementById('event_email_to');
-    recip.value = users
-  }
+    $('.alert').remove();
+    $('#event_email_to').val(users)
+    if(users.length == 0 ) {
+      $('#emailbody').prepend('#{error_message}');
+    }
+  }
```
Comment (bgeuken) [inline app/views/event_emails/new.html.haml]: I know, there was already inline js. But could you move that to the javascript assets? I would prefer to not have inline javascript.

File: app/views/event_emails/new.html.haml [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-        %a{:onclick => "user_email(#{users_for_event('incomplete')})"} Incomplete
-      %li
-        %a{:onclick => "user_email(#{users_for_event('submitted')})"} Submitted
-      %li
-        %a{:onclick => "user_email(#{users_for_event('approved')})"} Approved
-      %li
-        %a{:onclick => "user_email(#{users_for_event('canceled')})"} Cancel
+      - %w[All Accepted Incomplete Submitted Approved Cancel].each do |s|
+        %li
+          %a{id: 'state' + s, "data-users" => users_for_event(s.downcase) } #{s}
```
Comment (bgeuken) [inline app/views/event_emails/new.html.haml]: Nice that you cleaned that up :+1:

**Comment by ChrisBr** [comment]
Please add some screenshots :)

**Comment by nikhilgupta1211** [comment]
image
event_emails#new when no participants are present

**Comment by ChrisBr** [comment]
@bgeuken what do you think about the flash message? I had more in mind to set a hint into the recipient input field...

**Comment by bgeuken** [comment]
@ChrisBr I am not convinced this belongs to the input field. This is more about the recipients dropdown menu. When there are no event participants, the dropdown is disabled.

**Comment by ChrisBr** [comment]
@bgeuken Hmm, could also be an option!

**Comment by nikhilgupta1211** [comment]
I have pushed the changes

File: app/assets/javascripts/event_emails.js.coffee [hunk 3]
```diff
@@ -0,0 +1,9 @@
+if $('#select_recp').data('users') 
+	$('#select_recp').prop("disabled", true)
+
+states = (state) ->
+	$('#state'+state).click ->
+		users = $('#state'+state).data('users')
+		$('#event_email_to').val(users)
```
Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: Please don't use tabs for spacing. Just use whitespaces.

File: app/assets/javascripts/event_emails.js.coffee [hunk 4]
```diff
@@ -0,0 +1,9 @@
+if $('#select_recp').data('users') 
+	$('#select_recp').prop("disabled", true)
+
+states = (state) ->
```
Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: Please use a function name that reflects what the function does.

**Comment by bgeuken** [comment]
Please add the missing commit from #71 here. The other one includes the changes from this one anyway...

**Comment by nikhilgupta1211** [comment]
fixed it :)

**Pull request merged**